guides/Migrating Universal Modules to TypeScript.md

```diff
@@ -0,0 +1,91 @@
+# Migrating Universal Modules to TypeScript
+
+## Change default import to named import
+
+To optimize our libraries for _dead code elimination_ we should migrate our exports to be imported as such:
+
+```diff
+
+- import { FileSystem } from 'expo-file-system';
+
++ import * as FileSystem from 'expo-file-system';
+
+```
+
+Ideally we would make the main entry-point of a module be a file named like the module:
+`src/FileSystem.ts` (`build/FileSystem.js`).
+To migrate I propose we do what `expo-constants` does:
+
+**`src/index.ts`**
```

Comment:
This is useful only for modules that lots of people would have been using the module directly (not through the `expo` package). For almost all (or all?) of our unimodules, I suspect very few were used directly and making the breaking change is fine, plus we'll bump semver anyway.